networkx/algorithms/dag.py

```diff
@@ -139,6 +141,13 @@ def topological_sort(G, nbunch=None, reverse=False):
 
     if nbunch is None:
         nbunch = G.nodes_iter()
+        def direct_descendants(w):
+            return G[w]
+    else:
+        nbunch_index = {x: i for i, x in enumerate(nbunch)}
+        len_nbunch = len(nbunch)
+        def direct_descendants(w):
+            return sorted(G[w], key=lambda x: nbunch_index.get(x, len_nbunch))
     for v in nbunch:     # process all vertices in G
```

Comment:
This doesn't process all vertices in G.  Only those in nbunch.
Is it a topological sort if we only start at some of the nodes?